fh-sync-js.d.ts

```diff
@@ -426,12 +426,20 @@ declare module SyncClient {
    * @param handler - function that wraps underlying storage solution
    */
   function setStorageAdapter(handler: (dataset_id: string, isSave: boolean, cb: any) => void): void;
+
   /**
    * Allows to override default hashing method
    *
    * @param method - function that wraps underlying hashing method
    */
   function setHashMethod(method: () => any): void;
+
+  /**
+   * Allows to override default method for checking if application is online
+   *
+   * @param method - function that checks if network is available
+   */
+  function setNetworkStatusHandler(method: (callback: (isOnline: boolean) => void) => void): void;
```

Comment:
As I see. The name threw me. It's not really a handler for the network status, but a *provider* of the status. It provides sync with the network status.

---

Also, unrelated to this PR, but the method `setHashMethod` above this I think should take in `any` as the argument. Otherwise TS will give you a type error for something like `setHashMethod((x: any) => x.id);` because `(x: any) => any` is not assignable to `() => any`.